dask_cuda/proxy_object.py

```diff
@@ -155,15 +156,31 @@ def __init__(
         self._obj_pxy_lock = threading.RLock()
         self.__obj_pxy_cache = {}
 
-    def _obj_pxy_get_meta(self):
-        """Return the metadata of the proxy object.
+    def _obj_pxy_get_init_args(self, include_obj=True):
+        """Return the attributes needed to initialize a ProxyObject
+
+        Notice, the returned dictionary is ordered as the __init__() arguments
+
+        Parameters
+        ----------
+        include_obj: bool
+            Whether to include the "obj" argument or not
 
         Returns
         -------
-        Dictionary of metadata
+        Dictionary of attributes
         """
-        with self._obj_pxy_lock:
-            return {k: self._obj_pxy[k] for k in self._obj_pxy.keys() if k != "obj"}
+        args = [
+            "fixed_attr",
+            "type_serialized",
+            "typename",
+            "is_cuda_object",
+            "subclass",
+            "serializers",
+        ]
+        if include_obj:
+            args.insert(0, "obj")
+        return OrderedDict([(a, self._obj_pxy[a]) for a in args])
```

Comment:
Notice, even though `dict` is order as of Python 3.7, I think it make sense to use `OrderedDict` to make it explicit.